### PR TITLE
port TF Lite micro speech example to CEVA-DSP BX1

### PR DESCRIPTION
**Pull request by idog-ceva**
This commit is a port of TF Lite Micro Speech example to CEVA-DSP BX1 processor.
The commit adds project settings for CEVA's IDE as well as some code changes and additions, and demo input file.

### CODE REVIEW

**Comment by idog-ceva** [comment]
@googlebot I signed it!

**Comment by gbaned** [comment]
@idog-ceva Thank you for your contribution. Can you please sign CLA? Thanks!

**Comment by idog-ceva** [comment]
@googlebot I signed it!

**Comment by idog-ceva** [comment]
@googlebot I signed it!

File: tensorflow/lite/micro/examples/micro_speech/CEVA_BlueVox/main_functions_cpp.cc [hunk 1]
```diff
@@ -0,0 +1,306 @@
+
+
```
Comment (petewarden) [inline tensorflow/lite/micro/examples/micro_speech/CEVA_BlueVox/main_functions_cpp.cc:2]: Can you add the standard TensorFlow copyright notice here?
The file should also be formatted using clang-format -style=google

File: tensorflow/lite/micro/examples/micro_speech/CEVA_BlueVox/main_functions_cpp.cc [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+int16_t g_audio_capture_buffer[kAudioCaptureBufferSize];
+int16_t g_audio_output_buffer[kMaxAudioSampleSize];
+int32_t g_latest_audio_timestamp = 0;
+
+int16_t audio[513];
+int scores_count[4] = {0};
+
+
+
+char filename[50] = "input.wav";
```
Comment (petewarden) [inline tensorflow/lite/micro/examples/micro_speech/CEVA_BlueVox/main_functions_cpp.cc:30]: Can you break the file loading code into something like examples/micro_speech/ceva_bluevox/audio_provider.cc ?
That should help reduce the amount of duplicated code that's needed here.

File: tensorflow/lite/micro/examples/micro_speech/CEVA_BlueVox/main_functions_cpp.cc [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+
+// The name of this function is important for Arduino compatibility.
+void setup_tf() {
+  // Set up logging. Google style is to avoid globals or statics because of
+  // lifetime uncertainty, but since this has a trivial destructor it's okay.
+  // NOLINTNEXTLINE(runtime-global-variables)
+  static tflite::MicroErrorReporter micro_error_reporter;
+  int i;
+  error_reporter = &micro_error_reporter;
+#if 1
```
Comment (petewarden) [inline tensorflow/lite/micro/examples/micro_speech/CEVA_BlueVox/main_functions_cpp.cc:175]: Is this code needed anymore?

File: tensorflow/lite/micro/examples/micro_speech/simple_features/CEVA_BlueVox/fft.cc [hunk 4]
```diff
@@ -0,0 +1,99 @@
+
+
+
```
Comment (petewarden) [inline tensorflow/lite/micro/examples/micro_speech/simple_features/CEVA_BlueVox/fft.cc:3]: I don't believe this file is used, it's legacy, instead the one in micro_features implements the fft now used in the example.

**Review by petewarden — COMMENTED** [review]
Thanks for this! Comments below:

- Can you download libceva_dsp_bx1_lib.a from an external URL, since large binaries aren't allowed in the repo? You can see how to download in the makefile here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/tools/make/Makefile#L255
- Is tensorflow/lite/micro/tools/make/targets/CEVA_BlueVox/Debug/MICRO_SPEECH_DEMO.elf needed?
- tensorflow/lite/micro/tools/make/targets/CEVA_BlueVox/input.wav should be downloaded separately too.
- Can we move the makefile into a generated project, rather than hardcoding a single micro speech example? That would help us build all the examples, instead of just one.

**Comment by gbaned** [comment]
@idog-ceva Can you please check @petewarden's comments and keep us posted ? Thanks!

**Comment by yair-ehrenwald** [comment]
> @idog-ceva Can you please check @petewarden's comments and keep us posted ? Thanks!

@gbaned Hi, I'll take this over on CEVA's side and deal with the issues @petewarden raised in the next few days, thanks :)

**Comment by gbaned** [comment]
@yair-ehrenwald, @idog-ceva Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
@yair-ehrenwald, @idog-ceva Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
@yair-ehrenwald, @idog-ceva  Any update on this PR? and please resolve conflicts. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Comment by yair-ehrenwald** [comment]
> I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

Ok. I opened a new PR for this a couple of weeks ago:
https://github.com/tensorflow/tensorflow/pull/42414
this one should have been closed, sorry about that.

File: tensorflow/lite/micro/examples/micro_speech/CEVA_BlueVox/main_functions_cpp.cc (hunk 1 above)
Comment (rebeccaann95531) [reply tensorflow/lite/micro/examples/micro_speech/CEVA_BlueVox/main_functions_cpp.cc:2]: How do I do that I'm new at this

**Pull request closed without merging**